### PR TITLE
Remove last use of .expect() during Salus boot

### PR DESCRIPTION
**Pull request by rbradford**
This PR removes the last use of .expect() during the Salus boot replacing with error reporting.

- main: Use fmt::Display for errors that have it
- hyp_map, smp: Propagate error from stack memory allocation
- page-tracking: Propagate errors from hypervisor page allocator

See: #100

### CODE REVIEW

File: src/hyp_map.rs [hunk 1]
```diff
@@ -577,7 +590,7 @@ impl HypMap {
             .next()
             .unwrap();
```
Comment (glg-rv) [inline src/hyp_map.rs:591]: .expect() was added as an improvement to the unwrap() that was all over the initialization code.

I think in this respect this unwrap, and others if exist, should be equally handled.

In general, I'd like error happening during initialization to be clearly marked as fatal. Either in comments or names.

**Review by glg-rv — COMMENTED** [review]
Overall lgtm.

File: src/hyp_map.rs (hunk 1 above)
Comment (rbradford) [reply src/hyp_map.rs:591]: I agree, it would be a lot of work but it would be nice to know that there is no scope for panic.

Comment (glg-rv) [reply src/hyp_map.rs:591]: In this case would be a `ok_or` ? :-)

Comment (rbradford) [reply src/hyp_map.rs:591]: Yes `.ok_or` is a good solution for places like this.

**Review by sboeuf — APPROVED** [review]
:+1:

**Review by abrestic-rivos — APPROVED** [review]

File: src/hyp_map.rs (hunk 1 above)
Comment (glg-rv) [reply src/hyp_map.rs:591]: And hasn't been changed why?

**Pull request merged**